### ISSUE TITLE
recurrent network wrong GPU

### ISSUE THREAD

**Issue by ali-r** [pytorch/pytorch#76284]
### 🐛 Describe the bug

hello pytorch team

I have an issue with recurrent networks in pytorch
when i load the model it is in cuda:2 (i check the internal parametars of my GRU all in cuda with index 2)
but wen data pass the RNN network some memory from GPU-0 is consumed
and when GPU-0 is full it throw err

i think this question is related https://stackoverflow.com/questions/62067849/pytorch-model-training-runtimeerror-cudnn-error-cudnn-status-internal-error

this is not happen when CUDA_VISIBLE_DEVICES used

### Versions

PyTorch version: 1.10.0+cu102
Is debug build: False
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.4 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: Could not collect
CMake version: version 3.10.2
Libc version: glibc-2.25

Python version: 3.6.9 (default, Mar 15 2022, 13:55:28)  [GCC 8.4.0] (64-bit runtime)
Python platform: Linux-4.15.0-167-generic-x86_64-with-Ubuntu-18.04-bionic
Is CUDA available: True
CUDA runtime version: Could not collect
GPU models and configuration:
GPU 0: NVIDIA GeForce RTX 2080 Ti
GPU 1: NVIDIA GeForce RTX 2080 Ti
GPU 2: NVIDIA GeForce RTX 2080 Ti
GPU 3: NVIDIA GeForce RTX 2080 Ti

Nvidia driver version: 470.103.01
cuDNN version: /usr/lib/x86_64-linux-gnu/libcudnn.so.7.6.4
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.19.5
[pip3] pytorch-warmup==0.0.4
[pip3] torch==1.10.0+cu102
[pip3] torchaudio==0.10.0+cu102
[pip3] torchtext==0.4.0
[pip3] torchvision==0.11.1+cu102
[conda] Could not collect
[version.txt](https://github.com/pytorch/pytorch/files/8549357/version.txt)

**Comment by ngimel** [comment]
This is fixed on master/1.11, please reopen if still reproes.
